homeassistant/components/neato/vacuum.py

```diff
@@ -185,13 +185,17 @@ def update(self):
             self._mapdata[self._robot_serial]['maps'][0]['run_charge_at_end'])
 
         if self._robot_has_map:
+            robot_map_id = {}
             if self._state['availableServices']['maps'] != "basic-1":
                 if self._robot_maps[self._robot_serial]:
-                    robot_map_id = (
-                        self._robot_maps[self._robot_serial][0]['id'])
-
-                    self._robot_boundaries = self.robot.get_map_boundaries(
-                        robot_map_id).json()
+                    allmaps = self._robot_maps[self._robot_serial]
+                    for maps in allmaps:
+                        robot_map_id[allmaps.index(maps)] = maps['id']
+
+                    for map_id, neatomap in robot_map_id.items():
+                        neatomap = robot_map_id[map_id]
```

Comment:
We don't need this line now.

Reply:
I had originally thought to remove that line but for some reason my editor said `map_id` would become an unused variable so I had left it.  Will remove it now :) thanks again!